test/e2e/network/util.go

```diff
@@ -125,3 +126,37 @@ func CheckSCTPModuleLoadedOnNodes(f *framework.Framework, nodes *v1.NodeList) bo
 	}
 	return false
 }
+
+// execSourceIPTest executes curl to access "/clientip" endpoint on target address
+// from pause Pod to check if source ip is preserved.
+func execSourceIPTest(pausePod v1.Pod, targetAddr string) (string, string) {
+	var err error
+	var stdout string
+	timeout := 2 * time.Minute
+
+	framework.Logf("Waiting up to %v to get response from %s", timeout, targetAddr)
+	cmd := fmt.Sprintf(`curl -q -s --connect-timeout 30 %s/clientip`, targetAddr)
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(2 * time.Second) {
+		stdout, err = framework.RunHostCmd(pausePod.Namespace, pausePod.Name, cmd)
+		if err != nil {
+			framework.Logf("got err: %v, retry until timeout", err)
+			continue
+		}
+		// Need to check output because it might omit in case of error.
+		if strings.TrimSpace(stdout) == "" {
+			framework.Logf("got empty stdout, retry until timeout")
+			continue
+		}
+		break
+	}
+
+	framework.ExpectNoError(err)
+
+	// The stdout return from RunHostCmd is in this format: x.x.x.x:port or [xx:xx:xx::x]:port
```

Comment:
I suspect this test may fail in some scenarios with v6 cause of the `[ ]` :thinking: 

Something to watch out for in the future.

Reply:
looks like `net.SplitHostPort` would strip the brackets: https://play.golang.org/p/YoTX6bnpXas

Reply:
good to know!